code/script.js

```diff
@@ -220,106 +267,175 @@ const generateBoard = () => {
+  generateBoard();
+  // Invoke the function to add the player's name in the beginning
+  addPlayerName();
+};
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  // This variable stores what option group (category) the question belongs to.  
+  const category = questions.options[questions.selectedIndex].parentNode.label.toLowerCase();
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  // This variable that stores the actual value of the question we've selected.
+  const value = questions.value;
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
-}
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion; 
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
+  // Compare the currentQuestion details with the secretPerson person details to determine if we should keep or remove people based on that
+  // Invoke the function filterCharacters, passing in the value of the variable "keep"
 
+  if (secretPerson[category].includes(value)) {
+    filterCharacters(true);
+  } else {
+    filterCharacters(false);
   }
-}
+
+  // Increase the counter of hints by 1 and return it
+  countOfHints++;
+  return countOfHints;
+};
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (category === 'hair' || category === 'eyes' || category === 'habit') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+        `Yes, the person has ${value} ${category}! Keep all people that have ${value} ${category}.`
+      );
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
-    }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+        `No, the person doesn't have ${value} ${category}! Remove all people that have ${value} ${category}.`
+      );
```

Comment:
I like that you worked on making a cleaner code here with grouping together the values that can have the same alert message